### DISCUSSION TITLE
Collection throws error: "Can only flip string and integer values, entry skipped"

### DISCUSSION THREAD

**Discussion by martio** [laravel/framework#47362] [Q&A]
### Laravel Version

10.11.0

### PHP Version

8.2.4

### Description

Collection throws error: "Can only flip string and integer values, entry skipped":

```
        "message": "array_flip(): Can only flip string and integer values, entry skipped",
        "file": "/var/www/html/vendor/laravel/framework/src/Illuminate/Collections/Collection.php",
        "line": 421
```

### Steps To Reproduce

Sorting data...

**Comment by cosmastech** [comment]
```php
collect(['a' => []])->flip();
```
would be creating an array like:
```php
$myArray = [
    [] => 'a'
];
```
But this returns a TypeError as arrays only accept strings & ints as their keys (to my knowledge).
